auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -667,6 +710,25 @@ public FacebookBuilder setPermissions(@NonNull List<String> permissions) {
         public static final class TwitterBuilder extends Builder {
             public TwitterBuilder() {
                 super(TwitterAuthProvider.PROVIDER_ID);
+
+                try {
+                    //noinspection unused to possibly throw
+                    Class c = com.twitter.sdk.android.core.TwitterCore.class;
+                } catch (NoClassDefFoundError e) {
+                    throw new RuntimeException(
+                            "Twitter provider cannot be configured " +
+                                    "without dependency. Did you forget to add " +
+                                    "'com.twitter.sdk.android:twitter-core:VERSION' dependency?");
+                }
+
+                if (getApplicationContext().getString(R.string.twitter_consumer_key)
+                        .equals(UNCONFIGURED_CONFIG_VALUE)
+                        || getApplicationContext().getString(R.string.twitter_consumer_secret)
+                        .equals(UNCONFIGURED_CONFIG_VALUE)) {
+                    throw new RuntimeException("Twitter provider unconfigured. Make sure to add" +
```

Comment:
What about extracting hardcoded string values like this to `strings.xml`?

Reply:
Nah... you don't put developer error messages in user-facing strings. 😄